### ISSUE TITLE
🐛 [firebase_firestore] Enabling persistence crashes app

### ISSUE THREAD

**Issue by lukepighetti** [firebase/flutterfire#11242]
## Bug report

Enabling Firestore persistence crashes Flutter app when first snapshot query is made

### Steps to reproduce

```dart
void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await Firebase.initializeApp();
  final f = FirebaseFirestore.instance;
  f.settings = const Settings(persistenceEnabled: true); // if this runs
  f.collection('foo').snapshots().listen(print); // this crashes

  runApp(const MaterialApp(home: Material(child: Center(child: Text("Foo")))));
}
```

Platforms affected:

| platform | status  |
| -------- | ------ |
| iOS      | ❌     |
| Android  | ✅     |

```
2023-07-09 15:54:33.891153-0400 Runner[25427:28871128] -[NSNull intValue]: unrecognized selector sent to instance 0x1ba9b0020
2023-07-09 15:54:33.898752-0400 Runner[25427:28871128] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNull intValue]: unrecognized selector sent to instance 0x1ba9b0020'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000018040e7c8 __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x0000000180051144 objc_exception_throw + 56
	2   CoreFoundation                      0x000000018041d47c +[NSObject(NSObject) instanceMethodSignatureForSelector:] + 0
	3   CoreFoundation                      0x00000001804126c8 ___forwarding___ + 1308
	4   CoreFoundation                      0x0000000180414b4c _CF_forwarding_prep_0 + 92
	5   Runner                              0x0000000104cb338c -[FLTFirebaseFirestoreReader FIRFirestoreSettings] + 296
	6   Runner                              0x0000000104cb307c -[FLTFirebaseFirestoreReader readValueOfType:] + 1364
	7   Flutter                             0x000000010b1b1f18 _ZL9ReadValuePKv + 52
	8   Runner                              0x0000000104cb50ac -[FLTFirebaseFirestoreReader FIRFirestore] + 88
	9   Runner                              0x0000000104cb3044 -[FLTFirebaseFirestoreReader readValueOfType:] + 1308
	10  Flutter                             0x000000010b1b1f18 _ZL9ReadValuePKv + 52
	11  Flutter                             0x000000010b1b26fc FlutterStandardCodecHelperReadValueOfType + 784
	12  Runner                              0x0000000104cb312c -[FLTFirebaseFirestoreReader readValueOfType:] + 1540
	13  Flutter                             0x000000010b1b1f18 _ZL9ReadValuePKv + 52
	14  Runner                              0x0000000104cb3e50 -[FLTFirebaseFirestoreReader FIRQuery] + 64
	15  Runner                              0x0000000104cb3060 -[FLTFirebaseFirestoreReader readValueOfType:] + 1336
	16  Flutter                             0x000000010b1b1f18 _ZL9ReadValuePKv + 52
	17  Flutter                             0x000000010b1b26fc FlutterStandardCodecHelperReadValueOfType + 784
	18  Runner                              0x0000000104cb312c -[FLTFirebaseFirestoreReader readValueOfType:] + 1540
	19  Flutter                             0x000000010b1b1f18 _ZL9ReadValuePKv + 52
	20  Flutter                             0x000000010b1b0a98 -[FlutterStandardMethodCodec decodeMethodCall:] + 96
	21  Flutter                             0x000000010b1ae2cc ___ZL39SetStreamHandlerMessageHandlerOnChannelPU31objcproto20FlutterStreamHandler8NSObjectP8NSStringPU33objcproto22FlutterBinaryMessengerS_PU29objcproto18FlutterMethodCodecS_PU27objcproto16FlutterTaskQueueS__block_invoke + 68
	22  Flutter                             0x000000010a79fa40 ___ZN7flutter25PlatformMessageHandlerIos21HandlePlatformMessageENSt21_LIBCPP_ABI_NAMESPACE10unique_ptrINS_15PlatformMessageENS1_14default_deleteIS3_EEEE_block_invoke + 108
	23  libdispatch.dylib                   0x000000010678c594 _dispatch_call_block_and_release + 24
	24  libdispatch.dylib                   0x000000010678dd5c _dispatch_client_callout + 16
	25  libdispatch.dylib                   0x000000010679e808 _dispatch_main_queue_drain + 1316
	26  libdispatch.dylib                   0x000000010679e2d4 _dispatch_main_queue_callback_4CF + 40
	27  CoreFoundation                      0x0000000180372ca4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
	28  CoreFoundation                      0x000000018036d360 __CFRunLoopRun + 1956
	29  CoreFoundation                      0x000000018036c7a4 CFRunLoopRunSpecific + 584
	30  GraphicsServices                    0x0000000188ff7c98 GSEventRunModal + 160
	31  UIKitCore                           0x000000010ecf237c -[UIApplication _run] + 868
	32  UIKitCore                           0x000000010ecf6374 UIApplicationMain + 124
	33  Runner                              0x000000010498f79c main + 64
	34  dyld                                0x0000000105989fa0 start_sim + 20
	35  ???                                 0x0000000105badf28 0x0 + 4391100200
	36  ???                                 0xfb26800000000000 0x0 + 18097292890123272192
)
libc++abi: terminating with uncaught exception of type NSException
```

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.4 22F66 darwin-arm64, locale en-US)
    • Flutter version 3.10.5 on channel stable at /Users/lukepighetti/development/flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (4 weeks ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/lukepighetti/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.80.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.68.0

[✓] Connected device (4 available)
    • Pixel 4 (mobile)          • 99261FFAZ00F4G                       • android-arm64  • Android 13 (API 33)
    • iPhone 14 (16.2) (mobile) • 51B3F17C-CB32-4B1C-B7DA-C8C783B9349A • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)           • macos                                • darwin-arm64   • macOS 13.4 22F66 darwin-arm64
    • Chrome (web)              • chrome                               • web-javascript • Google Chrome 114.0.5735.198

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 3.0.5
Flutter SDK 3.10.5
vgl 1.0.0

dependencies:
- app_links 3.4.3 [flutter plugin_platform_interface flutter_web_plugins]
- blurhash_dart 1.2.1 [image]
- cached_network_image 3.2.3 [flutter flutter_cache_manager octo_image cached_network_image_platform_interface cached_network_image_web]
- clipboard 0.1.3 [flutter]
- cloud_firestore 4.8.1 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- collection 1.17.1
- confetti 0.7.0 [flutter vector_math]
- cupertino_icons 1.0.5
- figma_squircle 0.5.3 [vector_math flutter]
- firebase_analytics 10.4.3 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_auth 4.6.3 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.14.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_database 10.2.3 [firebase_core firebase_core_platform_interface firebase_database_platform_interface firebase_database_web flutter]
- firebase_storage 11.2.3 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- flutter 0.0.0 [characters collection js material_color_utilities meta vector_math sky_engine]
- flutter_blurhash 0.7.0 [flutter]
- go_router 8.2.0 [collection flutter flutter_web_plugins logging meta]
- google_sign_in 6.1.4 [flutter google_sign_in_android google_sign_in_ios google_sign_in_platform_interface google_sign_in_web]
- icons_plus 4.0.0 [flutter flutter_svg]
- image 4.0.17 [archive meta xml]
- image_cropper 4.0.1 [flutter image_cropper_platform_interface image_cropper_for_web]
- image_picker 0.8.9 [flutter image_picker_android image_picker_for_web image_picker_ios image_picker_linux image_picker_macos image_picker_platform_interface image_picker_windows]
- listenable_stream 1.1.0 [rxdart flutter]
- package_info_plus 4.0.2 [ffi flutter flutter_web_plugins http meta path package_info_plus_platform_interface win32]
- paginate_firestore 1.0.3+1 [flutter bloc flutter_bloc cloud_firestore provider]
- provider 6.0.5 [collection flutter nested]
- rxdart 0.27.7
- share_plus 7.0.2 [cross_file meta mime flutter flutter_web_plugins share_plus_platform_interface file url_launcher_web url_launcher_windows url_launcher_linux url_launcher_platform_interface ffi win32]
- shared_preferences 2.1.2 [flutter shared_preferences_android shared_preferences_foundation shared_preferences_linux shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- sign_in_with_apple 5.0.0 [flutter meta sign_in_with_apple_platform_interface sign_in_with_apple_web]
- simple_rich_text 2.0.49 [flutter url_launcher]
- sprung 3.0.1 [flutter]
- timeago 3.5.0 [intl]
- tuple 2.0.2
- url_launcher 6.1.11 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- uuid 3.0.7 [crypto]

dev dependencies:
- flutter_launcher_icons 0.13.1 [args checked_yaml cli_util image json_annotation path yaml]
- flutter_lints 2.0.1 [lints]
- flutter_native_splash 2.3.1 [args flutter flutter_web_plugins js html image meta path universal_io xml yaml]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters collection js matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]

dependency overrides:
- paginate_firestore 1.0.3+1 [flutter bloc flutter_bloc cloud_firestore provider]

transitive dependencies:
- _flutterfire_internals 1.3.3 [collection firebase_core firebase_core_platform_interface flutter meta]
- archive 3.3.7 [crypto path pointycastle]
- args 2.4.2
- async 2.11.0 [collection meta]
- bloc 8.1.2 [meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- cached_network_image_platform_interface 2.0.0 [flutter flutter_cache_manager]
- cached_network_image_web 1.0.2 [flutter flutter_cache_manager cached_network_image_platform_interface]
- characters 1.3.0
- checked_yaml 2.0.3 [json_annotation source_span yaml]
- cli_util 0.4.0 [meta path]
- clock 1.1.1
- cloud_firestore_platform_interface 5.15.1 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 3.6.1 [_flutterfire_internals cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- convert 3.1.1 [typed_data]
- cross_file 0.3.3+4 [js meta]
- crypto 3.0.3 [typed_data]
- csslib 1.0.0 [source_span]
- fake_async 1.3.1 [clock collection]
- ffi 2.0.2
- file 6.1.4 [meta path]
- file_selector_linux 0.9.2 [cross_file file_selector_platform_interface flutter]
- file_selector_macos 0.9.3 [cross_file file_selector_platform_interface flutter]
- file_selector_platform_interface 2.6.0 [cross_file flutter http plugin_platform_interface]
- file_selector_windows 0.9.3 [cross_file file_selector_platform_interface flutter]
- firebase_analytics_platform_interface 3.6.3 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_analytics_web 0.5.4+3 [_flutterfire_internals firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_auth_platform_interface 6.15.3 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 5.5.3 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser js meta]
- firebase_core_platform_interface 4.8.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.6.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_database_platform_interface 0.2.5+3 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_database_web 0.2.3+3 [firebase_core firebase_core_web firebase_database_platform_interface flutter flutter_web_plugins js]
- firebase_storage_platform_interface 4.4.3 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_storage_web 3.6.3 [_flutterfire_internals async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http js meta]
- flutter_bloc 8.1.3 [bloc flutter provider]
- flutter_cache_manager 3.3.0 [clock collection file flutter http path path_provider pedantic rxdart sqflite uuid]
- flutter_plugin_android_lifecycle 2.0.15 [flutter]
- flutter_svg 2.0.7 [flutter vector_graphics vector_graphics_codec vector_graphics_compiler]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- google_identity_services_web 0.2.1 [js meta]
- google_sign_in_android 6.1.18 [flutter google_sign_in_platform_interface]
- google_sign_in_ios 5.6.2 [flutter google_sign_in_platform_interface]
- google_sign_in_platform_interface 2.4.1 [flutter plugin_platform_interface quiver]
- google_sign_in_web 0.12.0+2 [flutter flutter_web_plugins google_identity_services_web google_sign_in_platform_interface http js]
- html 0.15.4 [csslib source_span]
- http 0.13.6 [async http_parser meta]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- image_cropper_for_web 2.0.0 [flutter flutter_web_plugins image_cropper_platform_interface js]
- image_cropper_platform_interface 4.0.0 [flutter plugin_platform_interface http]
- image_picker_android 0.8.7+3 [flutter flutter_plugin_android_lifecycle image_picker_platform_interface]
- image_picker_for_web 2.2.0 [flutter flutter_web_plugins image_picker_platform_interface mime]
- image_picker_ios 0.8.8 [flutter image_picker_platform_interface]
- image_picker_linux 0.2.1 [file_selector_linux file_selector_platform_interface flutter image_picker_platform_interface]
- image_picker_macos 0.2.1 [file_selector_macos file_selector_platform_interface flutter image_picker_platform_interface]
- image_picker_platform_interface 2.8.0 [cross_file flutter http plugin_platform_interface]
- image_picker_windows 0.2.1 [file_selector_platform_interface file_selector_windows flutter image_picker_platform_interface]
- intl 0.18.1 [clock meta path]
- js 0.6.7 [meta]
- json_annotation 4.8.1 [meta]
- lints 2.1.1
- logging 1.2.0
- matcher 0.12.15 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.2.0
- meta 1.9.1
- mime 1.0.4
- nested 1.0.0 [flutter]
- octo_image 1.0.2 [flutter flutter_blurhash]
- package_info_plus_platform_interface 2.0.1 [flutter meta plugin_platform_interface]
- path 1.8.3
- path_parsing 1.0.1 [vector_math meta]
- path_provider 2.0.15 [flutter path_provider_android path_provider_foundation path_provider_linux path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.27 [flutter path_provider_platform_interface]
- path_provider_foundation 2.2.3 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.11 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.0.6 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.7 [ffi flutter path path_provider_platform_interface win32]
- pedantic 1.11.1
- petitparser 5.4.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.4 [meta]
- pointycastle 3.7.3 [collection convert js]
- process 4.2.4 [file path platform]
- quiver 3.2.1 [matcher]
- share_plus_platform_interface 3.2.1 [cross_file flutter meta mime plugin_platform_interface path_provider uuid]
- shared_preferences_android 2.1.4 [flutter shared_preferences_platform_interface]
- shared_preferences_foundation 2.2.2 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.2.0 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.2.0 [flutter plugin_platform_interface]
- shared_preferences_web 2.1.0 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.2.0 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- sign_in_with_apple_platform_interface 1.0.0 [flutter plugin_platform_interface meta]
- sign_in_with_apple_web 1.0.1 [flutter flutter_web_plugins sign_in_with_apple_platform_interface js]
- sky_engine 0.0.99
- source_span 1.9.1 [collection path term_glyph]
- sqflite 2.2.8+4 [flutter sqflite_common path]
- sqflite_common 2.4.5+1 [synchronized path meta]
- stack_trace 1.11.0 [path]
- stream_channel 2.1.1 [async]
- string_scanner 1.2.0 [source_span]
- synchronized 3.1.0
- term_glyph 1.2.1
- test_api 0.5.1 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.2 [collection]
- universal_io 2.2.2 [collection meta typed_data]
- url_launcher_android 6.0.36 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.1.4 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.5 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.5 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.3 [flutter plugin_platform_interface]
- url_launcher_web 2.0.17 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.6 [flutter url_launcher_platform_interface]
- vector_graphics 1.1.7 [flutter vector_graphics_codec]
- vector_graphics_codec 1.1.7
- vector_graphics_compiler 1.1.7 [args meta path_parsing xml vector_graphics_codec]
- vector_math 2.1.4
- win32 5.0.4 [ffi]
- xdg_directories 1.0.0 [meta path process]
- xml 6.3.0 [collection meta petitparser]
- yaml 3.1.2 [collection source_span string_scanner]

```

</details>

---

**Comment by lukepighetti** [comment]
More context, it looks like it crashes no matter if persistence is set to true or false. This feels like it might be developer error

```
$firestore.settings = const Settings(persistenceEnabled: false); // this sets up a crash
```

```
$firestore.settings = const Settings(persistenceEnabled: true); // so does this
```

**Comment by lukepighetti** [comment]
related: https://github.com/firebase/flutterfire/issues/11172

**Comment by lukepighetti** [comment]
This was resolved by upgrading to `cloud_firestore: ^4.8.2`!
